### PR TITLE
Globals everywhere via singleton

### PR DESCRIPTION
**Pull request by Jade-GG**
Alternative version that uses a singleton, reducing the amount of Cache calls from many (very many) per request to just one

### CODE REVIEW

**Comment by Jade-GG** [comment]
@indykoning @royduin @BobWez98 @kevinmeijer97 this is the alternative version that uses a singleton for more efficiency.

File: src/RapidezStatamicServiceProvider.php [hunk 1]
```diff
@@ -19,14 +19,17 @@
 
 class RapidezStatamicServiceProvider extends ServiceProvider
 {
+    $this->app->singleton(StatamicDataComposer::class);
```
Comment (royduin) [inline src/RapidezStatamicServiceProvider.php:22]: Uh? This should be in a method somewhere?

**Comment by royduin** [comment]
Shouldn't be there a `compose()` method? See: https://laravel.com/docs/master/views#view-composers

File: src/RapidezStatamicServiceProvider.php (hunk 1 above)
Comment (Jade-GG) [reply src/RapidezStatamicServiceProvider.php:22]: whoops

**Pull request merged**